### PR TITLE
💥 [RUMF-951] merge rum-recorder into rum

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Have a single package for all RUM features

## Changes

The goal of this PR is more or less to replace the current `rum` package with `rum-recorder`. But because `rum-recorder` is now the official API, a few changes have been introduced:

* the `makeRumPublicApi` (in `rum-core`) function now requires a second argument, to inject all "recorder" features. It will be used in `rum-slim` to pass an empty implementation, and in `rum` to pass the actual implementation.
* the `manualSessionReplayRecordingStart` option have been removed, making it the default behavior
* adjust E2E tests and performance script

The browser-extension still references the rum-recorder. This should be changed in a separate PR.

## Testing

CI

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 1]
```diff
@@ -48,7 +51,22 @@ export type StartRum<C extends RumInitConfiguration = RumInitConfiguration> = (
 
 type StartRumResult = ReturnType<typeof startRum>
 
-export function makeRumPublicApi<C extends RumInitConfiguration>(startRumImpl: StartRum<C>) {
+export interface RecorderApi {
+  public: {
+    startSessionReplayRecording: () => void
+    stopSessionReplayRecording: () => void
+  }
+  onRumStart: (
```
Comment (bcaudan) [inline packages/rum-core/src/boot/rumPublicApi.ts:59]: Nesting those APIs in a `public` object seems odd to me.
Any reason to promote it that much?

Comment (BenoitZugmeyer) [reply packages/rum-core/src/boot/rumPublicApi.ts:59]: I made it so it is clear that we need to wrap those functions in `monitor`, and we can just [merge](https://github.com/DataDog/browser-sdk/pull/941/files#diff-f16464c555cdc03f7a8b03f1b312bb09ea41cd1acf4fa773629e880a5a1ea72bR217) it in the rest of the API.

Comment (bcaudan) [reply packages/rum-core/src/boot/rumPublicApi.ts:59]: yep, by not doing the merge, we could:
- let `rumPublicApi` wrap these functions in monitor
- simplify the naming in `recordingApi` and use the public name only on `rumPublicApi` side:
```
rum.startSessionReplayRecording()
recording.start()
```
wdyt?

Comment (BenoitZugmeyer) [reply packages/rum-core/src/boot/rumPublicApi.ts:59]: Sounds good to me

File: packages/rum-core/src/boot/rumPublicApi.ts [hunk 2]
```diff
@@ -110,23 +128,34 @@ export function makeRumPublicApi<C extends RumInitConfiguration>(startRumImpl: S
     isAlreadyInitialized = true
 
     function doStartRum(initialViewName?: string) {
```
Comment (amortemousque) [inline packages/rum-core/src/boot/rumPublicApi.ts:130]: What about moving doStartRum out of initRum as it starts to become bigger and bigger. 
Not directly related to this PR but, IMO, the use of closures with 3 levels of depth `makeRumPublicApi` -> `initRum` -> `doStartRum`, is hard to read.

File: README.md [hunk 3]
```diff
@@ -6,14 +6,13 @@ The browser SDK is used to collect logs and RUM data from the browser.
...
-| browser-rum-recorder | [![npm version][31]][32] | [![bundle size][33]][34] | [datadog-rum-recorder][35] | [![API][1]][37] [![product][2]][38] |
-| browser-rum-core     | [![npm version][51]][52] | [![bundle size][53]][54] |                            |
-| browser-core         | [![npm version][41]][42] | [![bundle size][43]][44] |                            |
+| Package          | npm                      | size                     | cdn                    | doc                                 |
+| ---------------- | ------------------------ | ------------------------ | ---------------------- | ----------------------------------- |
+| browser-logs     | [![npm version][01]][02] | [![bundle size][03]][04] | [datadog-logs][05]     | [![API][1]][07] [![product][2]][08] |
+| browser-rum      | [![npm version][11]][12] | [![bundle size][13]][14] | [datadog-rum][15]      | [![API][1]][17] [![product][2]][18] |
+| browser-rum-slim | [![npm version][21]][22] | [![bundle size][23]][24] | [datadog-rum-slim][25] | [![API][1]][27] [![product][2]][28] |
```
Comment (ruthnaebeck) [inline README.md:13]: bundle size image not working

File: test/e2e/lib/helpers/sdk.ts [hunk 4]
```diff
@@ -13,11 +13,11 @@ export async function flushEvents() {
...
   //
-  // The rum-recorder code uses a Web Worker to format the request data to be sent to the intake.
-  // Because all Worker communication is asynchronous, it cannot send its request during the
-  // "beforeunload" event, but a few milliseconds after. Thus, when navigating, if the future page
-  // loads very quickly, the page unload may occur before rum-recorder have time to send its last
-  // segment.
+  // The RUM session replay recorder code uses a Web Worker to format the request data to be sent to
+  // the intake.  Because all Worker communication is asynchronous, it cannot send its request
+  // during the "beforeunload" event, but a few milliseconds after. Thus, when navigating, if the
+  // future page loads very quickly, the page unload may occur before the recorder have time to send
```
Comment (ruthnaebeck) [inline test/e2e/lib/helpers/sdk.ts:19]:
```suggestion
  // future page loads very quickly, the page unload may occur before the recorder has time to send
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Docs review

File: packages/rum/src/boot/recorderApi.ts [hunk 5]
```diff
@@ -50,14 +50,8 @@ export function makeRecorderApi(startRecordingImpl: StartRecording): RecorderApi
...
-    public: {
-      startSessionReplayRecording: monitor(() => {
-        startSessionReplayRecordingStrategy()
-      }),
-      stopSessionReplayRecording: monitor(() => {
-        stopSessionReplayRecordingStrategy()
-      }),
-    },
+    start: () => startSessionReplayRecordingStrategy(),
+    stop: () => stopSessionReplayRecordingStrategy(),
```
Comment (bcaudan) [inline packages/rum/src/boot/recorderApi.ts:54]: what about simplify strategy naming as well?

**Review by bcaudan — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (BenoitZugmeyer) [reply README.md:13]: Thanks for the review! The package is not yet published to NPM, so this is expected. It should work once released/published.

**Review by amortemousque — APPROVED** [review]

**Pull request merged**